plugins/modules/net_tools/nmcli.py

```diff
@@ -648,19 +666,21 @@ def __init__(self, module):
         self.dhcp_client_id = module.params['dhcp_client_id']
         self.zone = module.params['zone']
 
-        if self.ip4:
-            self.ipv4_method = 'manual'
+        if self.method4:
+            self.ipv4_method = self.method4
         else:
```

Comment:
GitHub doesn't allow me to create suggestions spanning a block which contains deleted lines, so this one cannot be applied directly:
```suggestion
        elif self.ipv4:
            self.ipv4_method = 'manual'
        else:
            self.ipv4_method = None
```
That way you don't have a nested `if` block, which makes the whole construct a bit easier to read. (Also, same for the IPv6 variant below.)